### PR TITLE
Add logic to use the apk hash 

### PR DESCRIPTION
**Pull request by rachaprince**
Update the SDK to use the apk zip hash returned by the backend. Fall back to the map of the external code hash until we've backfilled the apk hash in our backend.

Also renames some functions and variables for clarity.

### CODE REVIEW

File: firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/CheckForNewReleaseClientTest.java [hunk 1]
```diff
@@ -311,34 +290,66 @@ public void handleNewReleaseFromClient_whenNewReleaseIsSameAsInstalledAab_return
...
+      checkForNewReleaseClient.extractApkHash(packageInfo);
+      checkForNewReleaseClient.extractApkHash(packageInfo);
       // check that calculateApkInternalCodeHash is only called once
       mockedReleaseIdentificationUtils.verify(
-          () -> ReleaseIdentificationUtils.calculateApkInternalCodeHash(any()));
+          () -> ReleaseIdentificationUtils.calculateApkHash(any()));
     }
   }
+
+  private AppDistributionReleaseInternal.Builder getTestNewRelease() {
```
Comment (rachaprince) [inline firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/CheckForNewReleaseClientTest.java:336]: I modified this test to use the pattern described in: https://testing.googleblog.com/2018/02/testing-on-toilet-cleanly-create-test.html

**Review by mannyjimenez0810 — APPROVED** [review]

**Pull request merged**